tck/src/main/java/ee/jakarta/tck/data/standalone/entity/EntityTests.java

```diff
@@ -2396,7 +2396,12 @@ public void testUpdateQueryWithoutWhereClause() {
             assertEquals(120, b3.height); // increased by factor of 2
         }
 
-        assertEquals(3, shared.removeAll());
+        var removeAllResult = shared.removeAll();
+
+        if (!type.isKeywordSupportAtOrBelow(DatabaseType.GRAPH)) {
+            //We don't have any guarantee of NoSQL, mainly on eventual consistency databases
+            assertEquals(3, removeAllResult);
+        }
```

Comment:
I noticed a few lines up that the test does invoke the following wait which Kyle added to allow for eventual consistency:
```
        TestPropertyUtility.waitForEventualConsistency();
```
Are you running the TCK with the option to set a wait time?
And if so, is the mechanism insufficient in some way, or does the database not even capable of counting a number of deletions?

If it isn't capable and will never give the right answer, the repository operation should be made to raise UnsupportedOperationException to alert the user of this, and the test case can trap for it, allowing that exception for those types of databases.

Reply:
it does make sense.

Reply:
About what I said, I was wrong.

Even with  `TestPropertyUtility.waitForEventualConsistency(),` it won't make sense because this behavior will not be guaranteed in real life or production. 

Because the result will vary depending on the amount of data, replica strategy, and number of database nodes.

In this case, I recommend you enhance the documentation to explain that this return will vary according to NoSQL databases, mainly for eventual consistency.

Reply:
Yes, documentation needs to be updated regardless. We have #782 for that. I added a checkbox to it for this.